config/variants/_base.js

```diff
@@ -0,0 +1,33 @@
+import * as _ from 'lodash';
+import * as childProcess from 'child_process';
+
+function execSync(command) {
+  return childProcess.execSync(command, {encoding: 'UTF-8'}).trim();
+}
+
+// We expose the commit sha for more straightforward build reproduction and
+// tracing.
+let FULL_COMMIT = execSync('git rev-parse HEAD');
+let SHORT_COMMIT = FULL_COMMIT.substr(0, 7);
+if (!_.isEmpty(execSync('git status --porcelain'))) {
+  FULL_COMMIT  = `${FULL_COMMIT}+`;
+  SHORT_COMMIT = `${SHORT_COMMIT}+`;
+}
+
+/**
+ * Base configuration that all variants extend.
+ */
+export default {
+  fullCommit:  FULL_COMMIT,
+  shortCommit: SHORT_COMMIT,
+  // We need a monotonically increasing sequence for build identifiers. # of
+  // commits is a great proxy for that, considering that the only things that
+  // actually care about a build number are for builds against master.
+  buildNum: null,
+
+  // The endpoint to hit for API requests.
+  apiEndpoint: null,
+
+  // The S3 bucket to sync built assets to.
+  s3Bucket: null,
```

Comment:
Do these `null` keys need to be declared here? It seems like they're mainly here so they can be centrally documented, but I have a sneaking suspicion people will forget to ever update this file, and will instead just update the individual config files directly.